File: 정렬/가장큰수.py

def solution(numbers):
    numbers = list(map(str, numbers))

    numbers.sort(key = lambda x: x* 3, reverse=True)
    return str(int(''.join(numbers)))

    return answer

# 문자열은 국어사전 순처럼 정렬**
# numbers의 원소는 0 이상 1,000 이하 >> 라는 조건 덕분에 x*3의 상태로 비교하면 되는 것.
# 즉, "666"이 "101010"보다 크다고 할 수 있는 것.

File: 20-2학기/정렬/가장큰수.java
package sorting;

import java.util.*;

public class 가장큰수 {
	public String solution(int[] numbers) {
		String answer = new String();
		/** 1 **/
		String str_numbers[] = new String[numbers.length];

		/** 2 **/
		for (int i = 0; i < str_numbers.length; i++) {
			str_numbers[i] = String.valueOf(numbers[i]); // int->String
		}

		/** 3 **/ //**이해필요
		Arrays.sort(str_numbers, new Comparator<String>() {
			@Override
			public int compare(String o1, String o2) {
				return (o2 + o1).compareTo(o1 + o2);
			} /// o2+o1랑 o1+o2비교해서 큰거 리턴
		});

		/** 4 **/
		if (str_numbers[0].startsWith("0")) {
			answer += "0";
		} else {
			for (int j = 0; j < str_numbers.length; j++) {
				answer += str_numbers[j];
			}
		}

		return answer;
	}
}